nix/command-output.nix

```diff
@@ -0,0 +1,32 @@
+{ pkgs, config, lib, name ? "default-output", ... }:
+with lib;
+{
+  options = {
+    name = mkOption {
+      default = name;
+      type = types.str;
+      description = "Name of the Command Output.";
+    };
+
+    script = mkOption {
+      default = null;
+      type = types.nullOr types.str;
+      #type = types.nullOr (types.either types.str types.path);
+      description = ''
+        Text of a script which will produce a JSON value.
+        <warning>Warning: This uses shell features and is potentially dangerous.</warning>
+        Environment variables: 
+        <envar>$out</envar> is a temp directory available for use.
```

Comment:
what's $out ?

Reply:
`out` is available as an environment variable in the script as temp directory.
https://github.com/NixOS/nixops/pull/1048/files#diff-6d7517db2e330d537da7846c6ab7f4c5R69